synapse/cortex.py

```diff
@@ -5197,17 +5226,73 @@ async def count(self, text, opts=None):
 
         return i
 
+    async def _getMirrorOpts(self, opts):
+        mirropts = copy.deepcopy(opts)
+        mirropts['nomirror'] = True
+        mirropts['nexsoffs'] = (await self.getNexsIndx() - 1)
+        mirropts['nexstimeout'] = self.querymirroropts.get('timeout')
+        return mirropts
+
+    async def _getMirrorProxy(self):
+        proxy = None
+        try:
+            proxy = await self.querymirror.proxy(timeout=3)
+        except (TimeoutError, s_exc.IsFini):
+            mesg = 'Unable to get proxy for query mirror, running locally instead.'
+            logger.warning(mesg)
+        return proxy
+
     async def storm(self, text, opts=None):
-        '''
-        '''
+
         opts = self._initStormOpts(opts)
 
+        if self.querymirror is not None and not opts.get('nomirror'):
+            extra = await self.getLogExtra(text=text)
+            logger.info(f'Offloading Storm query {{{text}}} to mirror.', extra=extra)
+
+            proxy = await self._getMirrorProxy()
+            mirropts = await self._getMirrorOpts(opts)
+
+            if proxy is not None:
+                try:
+                    async for mesg in proxy.storm(text, opts=mirropts):
+                        yield mesg
+                    return
+
+                except s_exc.TimeOut:
+                    mesg = 'Timeout waiting for query mirror, running locally instead.'
+                    logger.warning(mesg)
+
+        if (nexsoffs := opts.get('nexsoffs')) is not None:
```

Comment:
Need to document the `nomirror` option at a minimum. Documenting the `nexsoffs` and `nexstimeout` and saying "you probably don't need to worry about these ever" is probably okay too.